worldengine/draw.py

```diff
@@ -425,8 +395,8 @@ def draw_riversmap_on_file(world, filename):
 
 
 def draw_grayscale_heightmap_on_file(world, filename):
-    img = ImagePixelSetter(world.width, world.height, filename)
-    draw_grayscale_heightmap(world, img)
+    img = PNGWriter.grayscale_from_array(world.elevation['data'], filename, scale_to_range=True)
+    #draw_grayscale_heightmap(world, img)
```

Comment:
In general the idea is to have a function which do all the work on an "abstract image" (e.g., `draw_grayscale_heightmap`) and then a simple wrapper (`draw_grayscale_heightmap_on_file`). The reason is that the abstract image could either end up on an image file on disk OR it could be used to draw on the screen (as we do in worldengine-gui). This is the basic principle to follow to ensure that the user see on the screen what it will be later saved on disk.

If we need exceptions for performance reason I would just suggest to document that and keep them as exceptions.